default.json

```diff
@@ -81,7 +81,7 @@
       "versioning": "regex:^((?<compatibility>.*)-)?v(?<major>\\d+)(\\.(?<minor>\\d+))(\\.(?<patch>\\d+))$"
     },
     {
-      "matchUpdateTypes": ["major", "minor", "patch"],
+      "matchUpdateTypes": ["major", "minor", "patch", "digest"],
```

Comment:
This shouldn't be required, as the top level `commitMessage` [in this config](https://github.com/balena-io/renovate-config/blob/16a7061b5910cd9caabb062dedc5e6f07fe8631b/default.json#L71) should be overriding the top level `commitMessage` in the balena-os config.

Besides, we also don't generally want to include currentVersion and newVersion in a digest commit as the former often doesn't exist. Hence the above linked rule at the top of the config should be preferred.

Reply:
Okay, I see the issue now.

[This](https://github.com/balena-os/renovate-config/blob/c64793ea689760f62f0b0da10e42eec5200fe689/default.json#L32) is effectively a packageRule, so it is able to take precedent over [this](https://github.com/balena-io/renovate-config/blob/16a7061b5910cd9caabb062dedc5e6f07fe8631b/default.json#L71), even though the later config is applied second.

So in this repo, we need the following to correctly merge and overwrite the packageRule coming from balena-os.

```json
  "digest": {
    "commitBody": "Update {{depName}}\n\nChange-type: patch",
  },
```

So you were correct that the above would have solved it, but we don't need the versions for digest commits.